graphql_compiler/query_formatting/common.py

```diff
@@ -134,17 +141,37 @@ def deserialize_json_argument(name, expected_type, value):
             GraphQLDateTime: datetime.datetime with tzinfo=pytz.utc
             GraphQLFloat: float
             GraphQLDecimal: decimal.Decimal
-            GraphQLInt: six.integer_types, supporting long integers
-            GraphQLString: six.string_types
+            GraphQLInt: int
+            GraphQLString: str
             GraphQLBoolean: bool
-            GraphQLID: six.string_types
+            GraphQLID: str
     """
     try:
         return _deserialize_anonymous_json_argument(strip_non_null_from_type(expected_type), value)
     except (ValueError, TypeError) as e:
         raise GraphQLInvalidArgumentError("Error parsing argument {}: {}".format(name, e))
 
 
+def deserialize_multiple_json_arguments(
+    arguments: Dict[str, Any], expected_types: Dict[str, GraphQLScalarType]
+) -> Dict[str, Any]:
+    """Deserialize GraphQL arguments parsed from a json file.
+
+    Args:
+        arguments: dict mapping argument names to json serialized values.
+        expected_types: dict mapping argument names to expected GraphQL types.
+
+    Returns:
+        a dict mapping the argument names to their deserialized values. See the docstring of
+        deserialize_json_argument for more info on how arguments are deserialized.
+    """
+    validate_that_dicts_have_the_same_keys(arguments, expected_types)
```

Comment:
The way I expect this function to be used, this should be a user error. The error raised from inside this function is an `AssertionError` with a developer-friendly but not user-friendly message. I'd rather see "epxected arguments for ..." or "unnecessary arguments provided", with a `InvalidArgumentError`, or something like that.